server/src/main/resources/rules/rules.js

```diff
@@ -317,6 +317,21 @@ function createPool(pool) {
         return poolSet;
     };
 
+    // When pool is missing vcpu and consumer is virtual, pool should use cores
+    // instead so we calculate calculate vcpu based on cores and sockets
+    pool.calculateVcpu = function (consumer) {
```

Comment:
Does this function need to be attached to the specific pool?

Reply:
No. I was originally calling it from other place and left it as a pool method. Inlined.